style.md

```diff
@@ -0,0 +1,672 @@
+
+``` ts
+const myString: string = "You say \"goodbye\" and I say \"hello\".";
+```
+
+## Use concatenation when writing long string literals
+
+String literals that are too long to fit on a single line should be spread across multiple lines by concatenating many shorter string literals.
+
+``` ts
+const myLongStringLiteral = (
+    "This is a long string literal. In fact, it is so long, that it could " +
+    "not possibly fit on a single line and still be readable. " +
+    `Remember that the recommended maximum line length is ${MaxLineLength} ` +
+    "characters!"
+);
+```
+
+## Prefer braces around arrow function bodies
+
+In general, the bodies of arrow functions should always be enclosed within
+braces `{}`.
+
+``` ts
+myArray.map(value => {
+    return value + value;
+});
+```
+
+## Use seperate declarations
+
+Do not declare multiple variables together on the same line. Use separate declarations, with each declaration on its own line.
+
+``` ts
+const myFirstNumber: number = 0;
+const mySecondNumber: number = 0;
+```
+
+## Start all comments with a space
+
+In general, there should always be a single space character in between a comment's opening slashes `//` and the first character of the comment's text.
+
+``` ts
+// Note the space at the beginning of this comment!
+```
+
+## Prefer multiple single-line comments to multi-line block comments
+
+In general, long comments should be spread across multiple single line comments (i.e. `// comment`) rather than given in a single multi-line block comment (i.e. `/* comment */`).
+
+``` ts
+// This is a longer comment providing a lot of information about a function.
+// It describes in detail the function's purpose, its accepted input, and its
+// expected output. It's far too much information to fit on just one line.
+function myFunction(value: number): number {
+    return doStuffWith(value);
+}
+```
+
+## Prefer comments on their own line to trailing comments
+
+In general, comments should be on their own line, preceding the code that they apply to, instead of on the same line, immediately after the code that they apply to.
+
+``` ts
+// [Comment explaining the purpose of my variable]
+let myVariable: number = 0;
+```
+
+## Prefer "as" to type assertions
+
+Prefer syntax such as `value as Type` over `<Type> value`.
+([Why?](https://stackoverflow.com/a/33503842/4099022))
+
+``` ts
+const myValue: number = myOtherValue as number;
+```
+
+## Prefer Type[] to Array<Type>
+    
+Prefer type syntax such as `myArray: string[]` to `myArray: Array<string>`.
+    
+``` ts
+const myStringArray: string[] = ["Hello", "World"];
+```
+
+
+
+# Naming
+
+## Characters allowed in identifiers
+
+In general, dentifiers should be made up only of the characters `A` through `Z`, `a` through `z`, and `0` through `9`. Identifiers generally should not have an underscore `_` in their name, even if they identify private members.
```

Comment:
Underscores are sometimes used as unused variable/argument names for callback functions which don't use them.

Reply:
Can you provide a specific example? I don't remember ever running into this particular case with JS or TS.

Reply:
https://github.com/Talon1024/paledit/blob/master/src/app/gradient-model/gradient.ts#L98

Reply:
Underscore restriction remains in the form of `5.3.`. Please create an issue or a PR if you still think that this guideline needs to be changed or removed.